### ISSUE TITLE
Is it possible to sync privacy list between multiple devices for the same user when he blocks someone?

### ISSUE THREAD

**Issue by DemonionFF** [QuickBlox/quickblox-ios-sdk#861]
**New Issue Checklist**

 - [x] I have read the [Documentation](https://quickblox.com/developers/IOS)
 - [x] I have searched for a similar issue in the [project](https://github.com/QuickBlox/quickblox-ios-sdk/issues) and found none

**Environment details**

Info                           | Value                                       |
-------------------------------|---------------------------------------------|
iOS Version                    | 10.0                                    |                                          
Quickblox iOS SDK version      | 2.11                                  |
Xcode Version                  | Xcode 8.3                              |

When user X blocks user Y on device_1 is it possible to notify user's_X device_2 too.
In order to keep his privacy list up-to-date?
I suppose when I set privacy list on one device _QBChatDelegate_ methods should be called on my another device. So privacy list are not synced between devices.

Also is it possible to fetch and modify privacy list via _QBRequest_ with success and fail completions?
Because delegate API is not so convenient) And if chat is disconnected privacy list is not modified really but PrivacyListDidNotSet is not triggered. 

Thanks for looking into the issue.

**Comment by Raikerian** [comment]
Hi,

Privacy lists are stored on the server. So they are already synchronized. Perhaps you have some specific problem with them?

**Comment by DemonionFF** [comment]
I mean is there any delegate/completion for notifying another device of same user when privacy list did change?

**Comment by Pro100Andrey** [comment]
Hi @DemonionFF, Any updates ?

**Comment by DemonionFF** [comment]
We had it enabled, but on another device delegates don't triggering.

**Comment by Raikerian** [comment]
Which delegates exactly are not triggering?

**Comment by DemonionFF** [comment]
func chatDidReceive(_ privacyList: QBPrivacyList) {
        print("char did receive list: \(privacyList)")
        
        self.privacyList = privacyList
    }
    
    func chatDidNotReceivePrivacyList(withName name: String, error: Any?) {
        print("unable to fetch privacy list")
    }
    
    func chatDidSetPrivacyList(withName name: String) {
        print("did set privacy list with name")
    }
    
    func chatDidNotSetPrivacyList(withName name: String, error: Any?) {
        print(error)
    }
    
    func chatDidSetDefaultPrivacyList(withName name: String) {
        print("did set default privacy list")
    }
    
    func chatDidNotSetDefaultPrivacyList(withName name: String, error: Any?) {
        print("unable to set default privacy list")
    }

Any of these methods didn't triggering.
Delegate setted via method 

    QBChat.instance.addDelegate(self)

Just tested. On first device block/unblock working and on second nothing happened with the same user.

**Comment by Pro100Andrey** [comment]
Hi @DemonionFF, 
Could you please provide logs from first and second devices?
Thanks!

**Comment by Raikerian** [comment]
Seems like a bug, we will check it (this might require new method, but not sure for now).

**Comment by Pro100Andrey** [comment]
Hi @DemonionFF,

There are couple workarounds/solutions how to do it:

1) Send a system message to yourself, with some custom param, e.g. 'updated_privacy_list'=<name>
In this case this messages will be delivered to ALL active resources(devices).
On current device you will just ignore this message

2) If you use a privacy list only in UI - you can refresh your list when you open a screen with it.

In any cases, server already has its updated privacy list value and will handle everything properly.

Thanks.
